tools/triage/add_triage_signature.py

```diff
@@ -1103,6 +1103,81 @@ def _process_ticket(self, url, issue_key):
             self._update_triaging_ticket(report)
 
 
+class NameserverInClusterNetwork(ErrorSignature):
+    def __init__(self, *args, **kwargs):
+        super().__init__(
+            *args,
+            **kwargs,
+            function_impact_label="bad_nameserver",
+            comment_identifying_string="h1. Nameserver in internal network",
+        )
+
+    def _get_resolv_conf_files(self, url, base_path):
+        files = []
+        try:
+            triage_logs_tar = get_triage_logs_tar(triage_url=url, cluster_id=get_metadata_json(url)["cluster"]["id"])
+        except FileNotFoundError:
+            return files
+
+        # Fetch control-plane directory from the bootstrap log bundle
+        control_plane_dir = triage_logs_tar.get(f"{base_path}/control-plane/")
+        for node_dir in control_plane_dir.iterdir():
+            try:
+                node_ip = os.path.basename(node_dir)
+                # Fetch resolv.conf file for this host
+                resolvconf = triage_logs_tar.get(f"{base_path}/control-plane/{node_ip}/network/resolv.conf")
+                files.append(resolvconf)
+            except FileNotFoundError:
+                logger.error(f"could not find resolv.conf for {node_ip}")
+                continue
+
+        # Fetch resolv.conf from the bootstrap host
+        resolvconf = triage_logs_tar.get(f"{base_path}/bootstrap/network/resolv.conf")
+        files.append(resolvconf)
+
+        return files
+
+    def _check_ip_in_cidr(self, ip_str, cidr_str):
+        ip_address = ipaddress.ip_address(ip_str)
+        network = ipaddress.ip_network(cidr_str, strict=False)
+        return ip_address in network
+
+    def _get_nameservers(self, url, base_path):
+        ns = set()
+        for resolvconf in self._get_resolv_conf_files(url, base_path):
+            PATTERN = "^nameserver (.*)$"
```

Comment:
let's put it as class constant (under ``class NameserverInClusterNetwork(ErrorSignature):``). in here it's just weird